interview-questions.md

```diff
@@ -9,59 +9,55 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: `This` is used to reference an instance of an object.
 
+  Researched answer: The `this` keyword in JavaScript refers to the object it belongs to. The keyword will have different values depending on where it's used. If used in a method, `this` refers to the owner object. It's also worth noting that is has some differences between strict mode and non-strict mode. If using `this` in a function, in strict mode, `this` is undefined.
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: You can use React to build applications and create user interfaces.
 
+  Researched answer: React is a JavaScript component library used to create user interfaces for websites and mobile applications. React provides developers with the resources needed to create reusable UI components. A few reasons to consider using React are it's easy learning curve, massive community, ability to reuse components, Virtual DOMS, flexibility, and use of React Hooks.
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: The render() method.
 
+  Researched answer: The only lifecycle method that is required in a React class component is render(). All other methods are considered optional.
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: JSX is a syntax extension to JavaScript.
 
+  Researched answer: JSX is a syntax extension for JavaScript. Like React, it was developed by Facebook to extend the functionalities of HTML structures into JavaScript. A benefit to using JSX is that there's no requirement for separate HTML and JS code. You can also use JSX with other frameworks and libraries as well.
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: YARN - yet another resource negotiator
 
+  Researched answer: Yarn is a package manager that helps to manage project dependencies like NPM. Yarn is a fast, reliable, and secure dependency management tool that was developed by Facebook and is open-source. A benefit to using yarn is that it stores dependencies locally to avoid the download of these packages multiple times. When required, users can easily reinstall because it fetches from the disk when executing the `yarn add` command.
```

Comment:
Yes, yarn will connect dependencies. The yarn.lock, package.json, and node_nodules will be updated.